book/src/pong-tutorial/pong-tutorial-03.md

```diff
@@ -116,190 +116,328 @@ We're finally ready to implement the `PaddleSystem` in `systems/paddle.rs`:
-use amethyst::input::InputHandler;
+use amethyst::{
+    core::transform::Transform,
+    ecs::SystemBuilder,
+    input::{get_input_axis_simple, InputHandler},
+    prelude::*,
+};
 
 // You'll have to mark PADDLE_HEIGHT as public in pong.rs
 use crate::pong::{Paddle, Side, ARENA_HEIGHT, PADDLE_HEIGHT};
 
 pub struct PaddleSystem;
 
 impl System for PaddleSystem {
-    type SystemData = (
-        .write_component::<Transform>()
-       .read_component::<Paddle>(),
-        .read_resource::<InputHandler>(),
-    );
-
-    fn run(&mut self, (mut transforms, paddles, input): Self::SystemData) {
-        for (paddle, transform) in (&paddles, &mut transforms).join() {
-            let movement = match paddle.side {
-                Side::Left => input.axis_value("left_paddle"),
-                Side::Right => input.axis_value("right_paddle"),
-            };
-            if let Some(mv_amount) = movement {
-                if mv_amount != 0.0 {
-                    let side_name = match paddle.side {
-                        Side::Left => "left",
-                        Side::Right => "right",
-                    };
-                    println!("Side {:?} moving {}", side_name, mv_amount);
-                }
-            }
-        }
+    fn build(self) -> Box<dyn ParallelRunnable> {
+        Box::new(
+            SystemBuilder::new("PaddleSystem")
+                .with_query(<(&Paddle, &mut Transform)>::query())
+                .read_resource::<InputHandler>()
+                .build(move |_commands, world, input, query_paddles| {
+                    // -- todo --
+                }),
+        )
     }
 }
 # fn main() {}
 ```
 
 Alright, there's quite a bit going on here!
 
-We create a unit struct `PaddleSystem`, and with the `SystemDesc` derive. This
-is short for **System Descriptor**. In Amethyst, systems may need to access
-resources from the `World` in order to be instantiated. For each `System`, an
-implementation of the `SystemDesc` trait must be provided to specify the logic
-to instantiate the `System`. For `System`s that do not require special
-instantiation logic, the `SystemDesc` derive automatically implements the
-`SystemDesc` trait on the system type itself.
-
-Next, we implement the `System` trait for it with the lifetime of the components
-on which it operates. Inside the implementation, we define the data the system
-operates on in the `SystemData` tuple: `WriteStorage`, `ReadStorage`, and
-`Read`. More specifically, the generic types we've used here tell us that the
-`PaddleSystem` mutates `Transform` components, `.write_component::<Transform>`, it
-reads `Paddle` components,.read\_component::<Paddle>()`, and also accesses the `InputHandler`resource we created earlier, using the`Read\`
-structure.
-
-> For `InputHandler`, make sure the parameter type is the same
-> as the one used to create the `InputBundle` earlier.
+We create a unit struct `PaddleSystem` and implement the `System` trait for it.
```

Comment:
This section involved a lot of re-writing. Please review this and let me know if anything's inaccurate or missing.